app/src/main/java/by/evgen/an11pazniakhomework14/view/MainActivity.kt

```diff
@@ -0,0 +1,286 @@
+package by.evgen.an11pazniakhomework14.view
+
+import android.os.Bundle
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import by.evgen.an11pazniakhomework14.R
+import by.evgen.an11pazniakhomework14.homework15.data.*
+import by.evgen.an11pazniakhomework14.homework15.observer.Editor
+import by.evgen.an11pazniakhomework14.homework15.observer.UpdateDrinksListener
+import by.evgen.an11pazniakhomework14.homework15.util.CatalogClass
+import by.evgen.an11pazniakhomework14.homework15.util.OrderList
+import by.evgen.an11pazniakhomework14.homework15.util.getInfo
+import by.evgen.an11pazniakhomework14.old.CatDataClass
+import by.evgen.an11pazniakhomework14.old.createNewBarsik
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var result: String
+    private lateinit var drinkTextView: TextView
+    private lateinit var catalogTextView: TextView
+    private var iter = 0
+    private var catalogClass: CatalogClass<Spirits> = CatalogClass()
+//    private var isFirstClick = true
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        findFibCycle(20) //поиск Фибоначчи в цикле
+        findFibRecursion(20) //поиск Фибоначчи рекурсией
+        countButton() //настройка кнопки для подсчета к-ва нажатий
+        createAndShowCatButton() //настройка кнопки для создания котиков Барсиков
+//        showNextDrink() //работа кнопки "следующий"
+//        showPreviousDrink() //работа кнопки "предыдущий"
+        addNewDrink() //работа кнопки "добавить"
+//        delete() //работа кнопки "удалить
+        showCatalog() //функция, заполняющая поле "каталог"
+        printAllInfoFromSingleton()//функция, заполняющая поле "синглтон"
+    }
+
+    private fun findFibCycle(countFib: Int) {
+        val fibonacciRes = findViewById<TextView>(R.id.textview_cycle_result)
+        var count1 = 1
+        var count2 = 1
+        var countTemp: Int
+        result = "$count1 $count2 "
+
+        for (i in 0..(countFib - 3)) {
+            countTemp = count1 + count2
+            result += "$countTemp "
+            count1 = count2
+            count2 = countTemp
+        }
+        fibonacciRes.text = result
+    }
+
+    private fun findFibRecursion(countFib: Int) {
+        val fibonacciRes = findViewById<TextView>(R.id.textview_recursion_result)
+        var temp: Int
+        result = ""
+        for (i in 1..countFib) {
+            temp =
+                recursion(i) //вызов функции для рекурсивного поиска числа Фибоначчи для переданного аргумента
+            result += "$temp "
+        }
+        fibonacciRes.text = result
+    }
+
+    private fun recursion(countFib: Int): Int {
```

Comment:
something like:
private fun recursion(countFib: Int) = when (countFib) {
             0 -> 0
             1 -> 1
             else -> (recursion(countFib - 1) + recursion(countFib - 2))
         }

Reply:
Done. Thank you for advise.